frontend/src/components/WofButton.vue

```diff
@@ -0,0 +1,94 @@
+<template>
+  <button :class="`button button--${type}`" :style="`font-size: ${this.size}rem;`" @click="onClick()">
+    <wof-icon v-if="this.icon" :icon="this.icon" :size="this.size" class="button--icon" />
```

Comment:
Add computed property 'iconClass' which return 'button__icon' if slot is not empty.
If slot is empty, it should return empty string.

Reply:
What's the point?

Reply:
Button with only a icon has unnecessary margin and because of that it's not in the middle.

Reply:
It's not because of v-if

Reply:
It's because of class="button--icon" which is always set

Reply:
But this element is not rendered if the prop icon is not passed due to v-if.

Reply:
Icon should have a margin only if slot is not empty

Reply:
... did you try it out? Here's what I get when the icon is not passed:
![image](https://user-images.githubusercontent.com/24480246/141541059-e0d0956c-9d03-4cd7-b5ad-7f32c86d2272.png)

Reply:
And here's the HTML code rendered:
![image](https://user-images.githubusercontent.com/24480246/141541100-93e1e120-2350-4c3c-a935-21d0bdc2b47f.png)

Reply:
Slot - text inside button.
Icon - passed via props.

If slot is empty, icon should not have margin

Reply:
Oh, the other way around. Understood.

Reply:
DOne

Reply:
`computed: {
    iconClass() {
      if(this.$slots.default) {
        return 'button__icon';
      }
      return '';
    }
  }`

Reply:
Ok, now should be good. Forgot to call the **proper** name of the computed property. Seems to do what it's supposed to now (fingers crossed)